docs/book/command-params.md

```diff
@@ -0,0 +1,203 @@
+# Command Params
+
+In default Symfony Console commands it is possible to define
+[input arguments and options](https://symfony.com/doc/current/console/input.html).
+
+You can still do that, but we encourage using input parameters,
+to simply write interactive commands.
+
+## Using input parameters
+
+Internally input parameters behaves like options with the following modifications:
+
+- all parameters are optional by default,
+- in interactive mode, when getting value parameter value, and the value
+  is not provided with the command call, user will be asked to provide the value,
+- in non-interactive mode, default value is used instead, if the parameter value
+  is not supplied, exception will be thrown,
+- values provided in command call are validated the same way as question asked
+  to provide the value by user,
+- if value of the parameter is provided by user is set back in input options,
+  so it can be passed to the next [command in chain](commands-chain.md) (if defined),
+- there is no shortcut version of parameter, as we have for option.
```

Comment:
I was going to reply and suggest returning the created `InputParam` from `addParam()` to allow setting that information, but then realized that internally, you marshal information for an `InputOption` and add it to the definition when calling `addParam()`; the only way that would work is if instead of doing that, you modified `Application::run()` or `Application::doRun()` to first iterate over the parameters to add options to the definition. Which _might_ be a possibility, but definitely adds complexity. But it's complexity hidden from the user, which might be nice. That would allow usage like this:

```php
$param = $this->addParam('module', 'Module to create', InputParam::TYPE_BOOL);
$param->setShortcut('m');
$param->addValidators(['pattern' => '/^[A-Z][a-zA-Z0-9_]*$/']);
```

If that seems like too much overhead and/or too brittle, I think adding another optional parameter makes sense. It's one issue I have with the symfony/console API, but at least it would be consistent with it, and make parameters consistent with the other options.